templates/connect-inject-deployment.yaml

```diff
@@ -138,6 +138,38 @@ spec:
                 {{- if not (kindIs "invalid" $resources.requests.cpu) }}
                 -default-sidecar-proxy-cpu-request={{ $resources.requests.cpu }} \
                 {{- end }}
+
+                {{- if .Values.connectInject.initContainer }}
```

Comment:
This Helm [doc](https://helm.sh/docs/chart_best_practices/values/#flat-or-nested-values) is related.

Reply:
I don't think we need to handle that case since I think it would be unlikely a user does that and since I haven't seen other locations where we're doing it. But I don't feel too strongly about it.

Reply:
Since it provides some extra protection, I'm going to leave it for now.